tools/ds-identify

```diff
@@ -1009,27 +1009,17 @@ vmware_guestinfo_ovfenv_err() {
 }
 
 vmware_guestinfo_metadata() {
-    if [ "${1:-}" = "err" ]; then
```

Comment:
Dropping this needs more thought, the original commit 3a031a75e7 introduced the vmware_guestinfo_err as a fallback mechanism for failures in vmware-rpctool but I don't see how we can ever get down any logical path  to invoke vmware_guestinfo_err for metadata, userdata or vendordata. Instead of dropping these conditionals (and avoiding any callers for vmware_guest_info_err (userdata|metadata|vendordata) should we instead be checking the for a non-empty stderr output or a non-zero exit code from vmware-rpctool and determining whether we need to fallback to the _err condition processing.

Reply:
Dropping doesn't change current behavior though, and neither did 3a031a7 with regards to stderr. I think dropping it here makes it clear whats actually happening, and perhaps we go ahead with that and also open a new issue and ask akutz what he intended when he wrote it?

Reply:
Ok I'm good with this approach as-is. Thanks, after confirming that vmware_guestinfo is just a wrapper for trying `vmware-rpctool` and falling back to `vmtoolsd` on non-zero exit code, there is nothing here that warrants a followup for this case. +1 as ds-identify is just using `vmware_guestinfo_(metadata|userdata|vendordata)` as a smoke test for presence of active configuration from the Vmware datasource. If ds-id can't get that config from either `vmware-rpctool` or `vmtoolsd`, we won't detect the VMware DS which is desired behavior because this image/environment won't have permissions to view that config content.